lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -3873,10 +3876,14 @@ namespace Js
         JavascriptFunction ** builtinFuncs = library->GetBuiltinFunctions();
 
         library->AddMember(regexPrototype, PropertyIds::constructor, library->regexConstructor);
+        library->regexConstructorSlotIndex = 0;
+        Assert(regexPrototype->GetSlot(library->regexConstructorSlotIndex) == library->regexConstructor);
```

Comment:
Seeing the hard coding of these slot indices in the code for real now is making me reconsider.  It is rather clumsy looking isn't it?

I wouldn't block this PR on it, but it would be worth considering revisiting this to get the indices from the propertyid programatically instead.  Adding a new GetAccessorPropertyIndexes API probably won't be fun.